### PR TITLE
Add fallback URL for upstream tarball

### PR DESCRIPTION
**Pull request by xclaesse**
Some upstream servers are not reliable and randomly fail on build-all
CI. We can automatically upload the tarball to github release artifacts
as fallback.

enet, libwebsockets: Reupload to add fallback URL

### CODE REVIEW

File: tools/create_release.py [hunk 1]
```diff
@@ -151,6 +152,14 @@ def upload(self, path: Path, mimetype: str):
         response = requests.post(self.upload_url, headers=headers, params=params, data=path.read_bytes())
         response.raise_for_status()
 
+    def create_source_fallback(self):
+        response = requests.get(self.wrap_section['source_url'])
```
Comment (eli-schwartz) [inline tools/create_release.py:156]: Shouldn't we use meson subprojects download for this, which performs its own safety checks?

Comment (xclaesse) [reply tools/create_release.py:156]: Meson is not installed in release CI job and the url/checksum should already have been checked by the CI before merging. I think that should be good enough, no?

Comment (eli-schwartz) [reply tools/create_release.py:156]: Hmm, fair point.

File: tools/create_release.py [hunk 2]
```diff
@@ -151,6 +152,14 @@ def upload(self, path: Path, mimetype: str):
...
+        response.raise_for_status()
+        filename = Path(self.wrap_section['source_filename'])
+        filename.write_bytes(response.content)
+        self.upload(filename, 'application/zip')
```
Comment (eli-schwartz) [inline tools/create_release.py:160]: What is the content type even used for? Seems like a pointless aspect of GitHub 's API. They should not and probably do not care whether it's a zip or a tarball. So the fact that we get this wrong probably does not matter.

Comment (xclaesse) [reply tools/create_release.py:160]: I could use `mimetypes.guess_type()` but I don't think it matters.

**Pull request merged**